src/components/Settings.vue

```diff
@@ -51,22 +51,64 @@ export default {
     ImageProcessor,
   },
   data: () => ({
-    ImageCheck: 0,
+    imageCheck: 0,
     imageDelete: 0,
     imageUrl: '',
+    imageHeight: 64,
+    imageWidth: 64,
+    minEdgeLength: 64,
+    minWidth: 1,
+    maxWidth: 100,
+    minHeight: 1,
+    maxHeight: 100,
   }),
   methods: {
     onImageOpen(data) {
-      this.imageUrl = data;
+      // we briefly unpack the image here from its url to automatically
+      // determine what the proportions of the transformed image should be
+      const tmpImg = new Image();
+      const self = this;
```

Comment:
Is this necessary? Isn't `this` still the same inside the function?

Reply:
Whoops. I thought I was having an issue with getting data from inside the callback...

Guess not. Will fix.